[IMP] fieldservice: Minor changes

osi-scampbell opened the pull request:
When a user changes the Internal Reference on a FSM Location, the complete_name field is not updated. Since it is part of the complete_name field, I believe it should update when changed.

Ticket #1499

When a user changes the Template we only set Type if it is False. We also want to append to the existing description instead of overwriting it when copyting notes.

Ticket #1679

When a user pulls up a tree view, enables a group_by filter, selects a record, then selects a smartbutton, if that group_by field is not on the corresponding smart_button object then a KeyError will be thrown. This is resolved by removing the group_by key from the context dictionary after it is copied.

Ticket #1336

On FSO's when a user sets the Location, the directions are then copied into the Location Directions on the FSO properly. However, whenever the call_notes() is called, it duplicates the Location Directions. The only thing that should be in this field are the directions from the assigned FSM Location.

Ticket #1854

On FSO's users were moving the Kanban cards and not triggering the logic that is stored in the buttons that move the stage on the form view. We prevent this by adding a boolean field that is returned True only when the button is clicked. If it is not through the button then we will throw a UserError and move the Kanban card back to its original stage

Ticket #1481

max3903 commented on fieldservice/models/fsm_location.py:
```diff
@@ -256,18 +256,18 @@ def action_view_contacts(self):
-                action['context'].update({'active_ids': contacts.ids})
-                action['context'].update({'active_id': ''})
+                action['res_id'] = locations.id
             return action
```
@osi-scampbell I am not sure to follow you here: This is the method to view contacts from the location form view and based on the number of contacts, we display the form or list view of contacts... The variable name seems correct to me.

osi-scampbell replied:
@max3903 You are correct this variable is named improperly

brian10048 commented on fieldservice/models/fsm_order.py:
```diff
@@ -324,17 +334,18 @@ def copy_notes(self):
                         self.description = (self.equipment_id.notes + '\n ')
         if self.location_id:
             s = self.location_id.direction
+            self.location_directions
```
?

brian10048 replied:
= ?

brian10048 commented on fieldservice/models/fsm_order.py:
```diff
@@ -324,17 +334,18 @@ def copy_notes(self):
-            self.todo = self.template_id.instructions
+            self.todo += self.template_id.instructions
+        if self.description:
+            self.description = '\n' + old_desc
```
```suggestion
            self.description += '\n' + old_desc
```
?

brian10048 commented on fieldservice/models/fsm_order.py:
```diff
@@ -355,7 +366,8 @@ def _onchange_template_id(self):
             self.scheduled_duration = self.template_id.hours
             self.copy_notes()
-            self.type = self.template_id.type_id
+            if not self.type:
```
Why prevent changing the type if it was already set?
For me, if the template has a type it should be set on the order when the template is changed. New template means new type.  Unless the template has no type and the order already has a type, then we could keep the order type as set

osi-scampbell committed:
[IMP] Update Complete Name for Ref

[IMP] Keep Type and Description

[IMP] Clear Group By Context

[IMP] Typo + Location Direction Duplicate

[IMP] Prevent Kanban Completed Change

[IMP] Variable Name Change locations -> contacts

[IMP] Redundant Button

[IMP] Clean up Notes

[IMP] Cleanup Notes

osi-scampbell committed:
[IMP] Remove Team

osi-scampbell committed:
[IMP] Change Type From Template

max3903 commented on fieldservice/models/fsm_location.py:
```diff
@@ -253,21 +253,21 @@ def action_view_contacts(self):
         contact to show.
         '''
-        for location in self:
+        for contact in self:
```
@osi-scampbell `self` is a recordset of locations not contacts...

brian10048 commented on fieldservice/models/fsm_order.py:
```diff
@@ -306,37 +314,41 @@ def onchange_scheduled_duration(self):
         if self.template_id:
-            self.todo = self.template_id.instructions
+            if self.template_id.instructions:
+                self.todo += self.template_id.instructions
```
This is causing a duplicate of the same instructions or appends a new set of instructions leaving the previous template instructions, which seems incorrect to me.  With a new template comes a new set of instructions

And in the case of nothing in self.todo it raises error when trying to set a template on the form

 ` File "/home/odoo/build/OCA/field-service/fieldservice/models/fsm_order.py", line 346, in copy_notes
    self.todo += self.template_id.instructions
TypeError: unsupported operand type(s) for +=: 'bool' and 'str'`

osi-scampbell committed:
[IMP] Rewrite Instructions + Variable Name Change

max3903 committed:
Update fsm_order.py

patrickrwilson reviewed (approved):
Looks good.

max3903 commented:
@brian10048 Can we merge this one?

max3903 commented:
/ocabot merge minor
